content/api/commands/mount.md

```diff
@@ -0,0 +1,634 @@
+wrap your component in that provider in your component tests. This is a good use
+case to create a custom mount command that wraps your components for you.
+
+Below are a few examples that demonstrate how. These examples can be adjusted
+for most other providers that you will need to support.
+
+### React Router
+
+If you have a component that consumes a hook or component from
+[React Router](https://reactrouter.com/), you will need to make sure the
+component has access to a React Router provider. Below is a sample mount command
+that uses `MemoryRouter` to wrap the component. Setup props for `MemoryRouter`
+can be passed in the options param as well:
+
+```jsx
+import { mount } from '@cypress/react'
+import { MemoryRouter } from 'react-router-dom'
+
+Cypress.Commands.add('mountWithRouter', (component, options = {}) => {
+  const { routerProps = { initialEntries: ['/'] }, ...mountOptions } = options
+
+  const wrapped = <MemoryRouter {...routerProps}>{component}</MemoryRouter>
+
+  return mount(wrapped, mountOptions)
+})
+```
+
+Typings:
+
+```ts
+import { MountOptions, MountReturn } from '@cypress/react'
+import { MemoryRouterProps } from 'react-router-dom'
+
+declare global {
+  namespace Cypress {
+    interface Chainable {
+      /**
+       * Mounts a React node
+       * @param jsx React Node to mount
+       * @param options Additional options to pass into mount
+       */
+      mountWithRouter(
+        jsx: React.ReactNode,
+        options?: MountOptions & { routerProps?: MemoryRouterProps }
+      ): Cypress.Chainable<MountReturn>
+    }
+  }
+}
+```
+
+In this setup, you can pass in custom props for the `MemoryRouter` in the
+`options` param. Below is an example test that ensures an active link has the
+correct class applied to it by initializing the router with `initialEntries`
+pointed to a particular route:
+
+```jsx
+import { Navigation } from './Navigation'
+
+it('home link should be active when url is "/"', () => {
+  // No need to pass in custom initialEntries as default url is '/'
+  cy.mountWithRouter(<Navigation />)
+
+  cy.get('a').contains('Home').should('have.class', 'active')
+})
+
+it('login link should be active when url is "/login"', () => {
+  cy.mountWithRouter(<Navigation />, {
+    routerProps: {
+      initialEntries: ['/login'],
+    },
+  })
+
+  cy.get('a').contains('Login').should('have.class', 'active')
+})
+```
+
+### Redux
+
+To use a component that consumes state or actions from a
+[Redux](https://react-redux.js.org/) store, you can create a `mountWithRedux`
+command that will wrap your component in a Redux Provider:
+
+```jsx
+import { mount } from '@cypress/react'
+import { Provider } from 'react-redux'
+import { getStore } from '../../src/store'
+
+Cypress.Commands.add('mountWithRedux', (component, options = {}) => {
+  // Use the default store if one is not provided
+  options.reduxStore = options.reduxStore || getStore()
```

Comment:
I'm guessing we don't want to pass `reduxStore` into `mount`. In that case,

```js
const { reduxStore = getStore(), ...mountOptions } = options
```

Reply:
i was looking to avoid destructuring syntax, but ya, makes sense here.

Reply:
done in bcc884c